internal/imagemgmt/imagemgmt.go

```diff
@@ -0,0 +1,278 @@
+package imagemgmt
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"syscall"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift-kni/lifecycle-agent/lca-cli/ops"
+	"github.com/samber/lo"
+
+	lcautils "github.com/openshift-kni/lifecycle-agent/utils"
+)
+
+// ImageMgmtIntf is an interface for LCA iamge management commands.
+//
+//go:generate mockgen -source=imagemgmt.go -package=imagemgmt -destination=mock_imagemgmt.go
+type ImageMgmtIntf interface {
+	CheckDiskUsageAgainstThreshold(thresholdPercent int) (bool, error)
+	GetInuseImages() (inUse []string, rc error)
+	GetPinnedImages() (pinned []string, rc error)
+	GetRemovalCandidates() (removalCandidates []imageMgmtImageInfo, rc error)
+	CleanupUnusedImages(thresholdPercent int) error
+}
+
+type ImageMgmtClient struct {
+	Log         *logr.Logger
+	Executor    ops.Execute
+	StoragePath string
+}
+
+// NewImageMgmtClient creates and returns a ImageMgmtIntf interface for image management commands
+func NewImageMgmtClient(log *logr.Logger, hostCommandsExecutor ops.Execute, storagePath string) ImageMgmtIntf {
+	return &ImageMgmtClient{
+		Log:         log,
+		Executor:    hostCommandsExecutor,
+		StoragePath: storagePath,
+	}
+}
+
+// imageMgmtImageInfo struct provides the image data required for image management
+type imageMgmtImageInfo struct {
+	Id          string
+	RepoTags    []string
+	RepoDigests []string
+	Names       []string
+	Created     uint64
+	Dangling    bool
+}
+
+// Types for image data from crictl commands
+type crictlContainerImageData struct {
+	Image string `json:"image"`
+}
+type crictlContainerData struct {
+	Image crictlContainerImageData `json:"image"`
+}
+
+type crictlContainersList struct {
+	Containers []crictlContainerData `json:"containers"`
+}
+
+type crictlImageData struct {
+	Id     string `json:"id"`
+	Pinned bool   `json:"pinned"`
+}
+
+type crictlImagesList struct {
+	Images []crictlImageData `json:"images"`
+}
+
+// Types for image data from podman commands
+type podmanContainerData struct {
+	ImageID string
+}
+
+// Use a var for syscall.Statfs in order to override it in mock tests
+var syscallStatfs = syscall.Statfs
+
+// CheckDiskUsageAgainstThreshold gets the disk usage for the specified path and checks whether it exceeds the specified threshold
+func (c *ImageMgmtClient) CheckDiskUsageAgainstThreshold(thresholdPercent int) (bool, error) {
+	var stat syscall.Statfs_t
+	if err := syscallStatfs(c.StoragePath, &stat); err != nil {
+		return false, fmt.Errorf("statfs failed for %s: %w", c.StoragePath, err)
+	}
+
+	// Get disk usage percentage
+	usedpct := int(((stat.Blocks - stat.Bavail) * 100) / stat.Blocks)
+	c.Log.Info(fmt.Sprintf("DEBUG: Disk usage of %s is currently %d%%", c.StoragePath, usedpct))
```

Comment:
C.Log.Debug?

Reply:
Only Info and Error are supported, I believe